CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,187 @@
+   * Makes the exchange:
+     * If selling CELO and purchasing stable token:
+       * The CELO is sent to the Reserve.
+       * Stable token is minted to the exchange proposer according to the rate originally recorded when the exchange was proposed.
+     * If selling stable token and purchasing CELO:
+       * The stable token is burned.
+       * CELO is transferred from the Reserve to the exchange proposer according to the rate originally recorded when the exchange was proposed.
+   * Records:
+     * The exchange as in the Executed state.
+5. **`function setStableTokenExchangeLimits(address stableToken, uint256 minExchangeAmount, uint256 maxExchangeAmount) external`**
+   * Only callable by Governance.
+   * Sets the minimum and maximum amounts of stable token that can be minted/burned in an exchange by updating `stableTokenExchangeLimits[stableToken]`.
+     * Effectively "enables" exchanging a stable token if the min & max amounts were previously 0.
+     * Effectively "disables" exchanging a stable token if setting the min & max amounts to 0.
+
+## Rationale
+
+Granda Mento is not intended to undermine the market or provide a way for large players to "dump" CELO for cXXX. It's also not intended to be an exceptionally elegant or complicated solution-- instead, this is looking to be something that is simple, straightforward, and works alongside the existing implementation of Mento. Any larger-scale Mento design modifications to facilitate large stable token mints are out of scope, and should be considered for a medium/long term design change to the stability protocol. The proposed implementation attempts to navigate the tradeoffs of complexity, centralization, economic safety, and ease of use for all actors in the Celo ecosystem.
+
+Some ideas other than the proposed approach include:
+
+1. Exchanges have a delay and require a deposit of the sold asset. Any exchange must be explicitly approved by a governance proposal to be executed.
+   * This relies upon the community as the arbiter of what exchanges are "safe."
+   * This could exacerbate the already-present voter fatigue.
+2. Large-volume exchanges could be made by anyone without Governance approval.
+   * The most compelling approach to this would involve an auction. This is complicated to implement, requires careful design, and was intentionally removed from the original Mento design.
+
+Approach (2) was deemed risky and complicated. Approach (1) felt more in line with the goal of having Granda Mento be simple and safe, but ultimately the involvement of Governance for every exchange felt unsustainable and exhausting for voters. The proposed implementation involving an exchange delay, a multisig for approvals, and Governance in the (likely rare) event the community wishes to veto an exchange, aims to relieve voter exhaustion while still providing the security of community scrutiny.
```

Comment:
This requires operational monitoring of the `GrandaMento` exchanges for governance participants
Could this lead to even more fatigue?

Reply:
Hmmm interesting point. In my opinion the fatigue around governance is less about being active in the community (eg discussing proposals or keeping up to date) but more around bigger players having to go through key management processes to use their voting keys frequently. So while there will be some responsibility on Governance now to keep an eye on exchange proposal approvals, I don't think it will dramatically affect fatigue

Reply:
I think it'll be the opposite, more proposals to keep track off means coordination and participation from others, even if it's not necessarily be the same as the regular governance process.